egs/timit/ASR/local/prepare_lexicon.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+# Copyright    2021  Xiaomi Corp.        (authors: Mingshuang Luo)
+#
+# See ../../../../LICENSE for clarification regarding multiple authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+This script takes as input supervisions json dir "data/manifests"
+consisting of supervisions_TRAIN.json and does the following:
+
+1. Generate lexicon.txt.
+
+"""
+import argparse
+import json
+import logging
+from pathlib import Path
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--manifests-dir",
+        type=str,
+        help="""Input directory.
+        """,
+    )
+    parser.add_argument(
+        "--lang-dir",
+        type=str,
+        help="""Output directory.
+        """,
+    )
+
+    return parser.parse_args()
+
+
+def prepare_lexicon(manifests_dir: str, lang_dir: str):
+    """
+    Args:
+      manifests_dir:
+        The manifests directory, e.g., data/manifests.
+      lang_dir:
+        The language directory, e.g., data/lang_phone.
+
+    Return:
+      The lexicon.txt file and the train.text in lang_dir.
+    """
+    phones = set([])
+
+    supervisions_train = Path(manifests_dir) / "supervisions_TRAIN.json"
+    lexicon = Path(lang_dir) / "lexicon.txt"
+
+    logging.info(f"Loading {supervisions_train}!")
+    with open(supervisions_train, "r") as load_f:
+        load_dicts = json.load(load_f)
+        for load_dict in load_dicts:
+            text = load_dict["text"]
+            # list the phone units and filter the empty item
+            phones_list = list(filter(None, text.split()))
```

Comment:
```suggestion
            phones_list = text.split()
```

Reply:
Em...this maybe keep it. Some texts are like follow：
`sil ay f  g sil`--->`['sil', 'ay', 'f', '', 'g', 'sil']`, so we have to filter this guy `''` .